iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -216,8 +216,8 @@ inline bool string<Capacity>::unsafe_assign(const char* const str) noexcept
     const uint64_t strSize{strnlen(str, Capacity + 1U)};
     if (Capacity < strSize)
     {
-        std::cerr << "Assignment failed. The given cstring is larger (" << strSize << ") than the capacity ("
-                  << Capacity << ") of the fixed string." << std::endl;
+        IOX_LOG(WARN) << "Assignment failed. The given cstring is larger (" << strSize << ") than the capacity ("
+                      << Capacity << ") of the fixed string.";
         return false;
```

Comment:
I think when the return value already indicates that something went wrong, this could be a `IOX_LOG(DEBUG)`. @FerdinandSpitzschnueffler since you implemented the string, what do you think.

Reply:
@elBoberido Good point, I agree.

Reply:
@elBoberido @FerdinandSpitzschnueffler Makes sense, will remember that. Changed all levels to `DEBUG`.

Reply:
@ithier oh, you changed it everywhere? I meant only the functions which communicate the failure with a return value